lwgenerate/aws.go

```diff
@@ -0,0 +1,294 @@
+package lwgenerate
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/pkg/errors"
+)
+
+type ExistingIamRoleDetails struct {
+	// Existing IAM Role ARN
+	Arn string
+
+	// Existing IAM Role Name
+	Name string
+
+	// Existing IAM Role External Id
+	ExternalId string
+}
+
+type GenerateAwsTfConfigurationArgs struct {
+	// Should we configure cloudtrail integration in LW?
+	ConfigureCloudtrail bool
+
+	// Should we configure CSPM integration in LW?
+	ConfigureConfig bool
+
+	// Supply an AWS region for where to find the cloudtrail resources
+	// TODO @ipcrm future: support split regions for resources (s3 one place, sns another, etc)
+	AwsRegion string
+
+	// Supply an AWS Profile name for the main account, only asked if configuring multiple
+	AwsProfile string
+
+	// Existing S3 Bucket ARN (Required when using existing cloudtrail)
+	ExistingCloudtrailBucketArn string
+
+	// Optionally supply existing IAM role details
+	ExistingIamRole *ExistingIamRoleDetails
+
+	// Existing SNS Topic
+	ExistingSnsTopicArn string
+
+	// Consolidated Trail
+	ConsolidatedCloudtrail bool
+
+	// Should we force destroy the bucket if it has stuff in it? (only relevant on new Cloudtrail creation)
+	ForceDestroyS3Bucket bool
+
+	// For AWS Subaccounts in consolidated CT setups
+	// TODO @ipcrm future: what about many individual ct/config integrations together?
+	Profiles map[string]string
+
+	// Lacework Profile to use
+	LaceworkProfile string
```

Comment:
Do all these fields need to be public?

Reply:
They'll have to be unless it would be better to create a constructor with getter/setters for all of these

Reply:
No longer public in the rework